text/0004-structured_data_io.md

```diff
@@ -100,6 +130,51 @@ Entries are themselves a small abstraction that carries the `Value` (parsed
 data), `Index` and `Key`. Only one of `Index` and `Key` will be populated at a
 given point, depending on weather an array or object is being read.
```

Comment:
Add something to the effect of "`Value` is of an arbitrary structure", to cut off any expectations that it is merely a primitive type.